src/dbnode/storage/limits/query_limits.go

```diff
@@ -190,11 +201,25 @@ func (q *lookbackLimit) exceeded() error {
 }
 
 func (q *lookbackLimit) checkLimit(recent int64) error {
-	if q.options.Limit > 0 && recent > q.options.Limit {
+	q.overrideLock.RLock()
```

Comment:
I think it would be cleaner to have a single `limit` var that can be set by either the config/constructor or by an overriding endpoint. less error prone.

Reply:
Makes sense. One question though - which behavior is better: (a) if you `Override(nil)` (i.e. remove your override) the limit returns to the original one? or (b)original val is lost and `Override` is the truth going forward. I think I lean (a) so that removing override returns to original behavior.

Reply:
hmm that's a valid point. maybe keep your impl

Reply:
Went w/ a single `limit` var as suggested but kept the behavior the same and just added a `func getLimit(opts LookbackLimitOptions, override *int64) *int64` func